### PR TITLE
Upgrade to Go v1.17

### PR DESCRIPTION
**Pull request by Ashmita152**
Signed-off-by: Ashmita Bohara <ashmita.bohara152@gmail.com>

## Which problem is this PR solving?
Upgrade to Go v1.17

## Short description of the changes
Ran `go mod tidy` too

### CODE REVIEW

**Comment by Ashmita152** [comment]
https://github.com/golang/go/issues/47810

**Review by yurishkuro — COMMENTED** [review]
I think the version is also present in docker files.

File: cmd/agent/app/servers/thriftudp/socket_buffer.go [hunk 1]
```diff
@@ -12,6 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//go:build !windows
 // +build !windows
```
Comment (yurishkuro) [inline cmd/agent/app/servers/thriftudp/socket_buffer.go:15]: Do we really need both directives?

File: docker/Makefile [hunk 2]
```diff
@@ -1,7 +1,7 @@
 VERSION := 1.0.0
 ROOT_IMAGE ?= alpine:3.13
 CERT_IMAGE := $(ROOT_IMAGE)
-GOLANG_IMAGE := golang:1.16-alpine
+GOLANG_IMAGE := golang:1.17-alpine3.13
```
Comment (yurishkuro) [inline docker/Makefile:4]: Is there a ticket that is pending? If so, please add it to a TODO comment.

**Comment by Ashmita152** [comment]
I think this handles all usages of golang 1.16 to 1.17.

**Review by yurishkuro — APPROVED** [review]
Thanks!

**Pull request merged**